File: src/Java/Creational Patterns/Factory/CreditCards/src/Creators/PlatinumCreditCardCreator.java
package Creators;

import Products.CreditCard;
import Products.Platinum;

import java.math.BigDecimal;

public class PlatinumCreditCardCreator implements ICreditCardCreator {
    @Override
    public CreditCard createCreditCard(String model, BigDecimal limit, BigDecimal annualCharge) {
        return new Platinum(model, limit, annualCharge);
    }
}

File: src/JavaScript/CreationalPatterns/Factory/CreditCards/creators/platinumCreditCardCreator.js

const ICreditCardCreator = require('./creditCardCreator');
const Platinum = require('../products/platinum');

/**
 * @implements {ICreditCardCreator}
 */
class PlatinumCreditCardCreator {
    /**
     * @inheritDoc
     */
    createCreditCard(model, limit, annualCharge) {
        return new Platinum(model, limit, annualCharge);
    }
}

module.exports = PlatinumCreditCardCreator;